smpp/server.go

```diff
@@ -127,6 +125,21 @@ func NewUnstartedServer(user, password string, listener net.Listener) Server {
 	return s
 }
 
+// NewTLDUnstartedServer creates a new SSL Server with default settings, and
+// does not start it. Callers are supposed to call Start and Close later.
+func NewTLSUnstartedServer(user, password string, listener net.Listener, tlsConf *tls.Config) Server {
+	s := &server{
+		User:   user,
+		Passwd: password,
+		TLS:    tlsConf,
+		m:      map[pdu.ID]RequestHandlerFunc{},
+		a:      map[pdu.ID]AuthRequestHandlerFunc{},
+		s:      map[string]Session{},
+		l:      listener,
+	}
+	return s
+}
+
```

Comment:
Yeah, this server started with fiorix as just a mock Server and we have been modifying to our needs into a more production server. You can look at `conn.go`, I think TLS is used there for the client.